idaes/tests/test_import.py

```diff
@@ -11,81 +11,285 @@
+            stdout=subprocess.PIPE,
+            timeout=max_time,
+        )
+    except subprocess.CalledProcessError as err:  # total failure!
+        result.error = str(err)
+    except subprocess.TimeoutExpired:  # timeout
+        result.error = f"Timeout (> {max_time}s)"
+
+    if proc is not None:
+        output = proc.stdout.decode()
+        if proc.returncode != 0:
+            # find "traceback" text in output
+            result.error = "Non-zero return code:\n"
+            for line in output.splitlines():
+                if not line.startswith("import time:"):
+                    result.add_error(line)
+        else:
+            # Parse output of '-X importtime'
+            prev_import = None
+            for line in output.splitlines():
+                # For rows with timings
+                if line.startswith("import time:"):
+                    # Save if match with current module
+                    if line.endswith(module):
+                        times = re.findall(r"\d+", line)
+                        result.timing = float(times[1]) / 1e6
+                    # In any case, remember module name
+                    prev_import = line[line.rfind("|") + 1:].strip()
+                elif not allow_output:
+                    # Associate dis-allowed output with last seen module name
+                    result.add_error(f"[{prev_import}] {line}")
+
+    # Return timing and/or error information
+    return result
+
+
+####################################################################################
 
 
 @pytest.mark.unit
-def test_import():
-    root_dir = pathlib.Path(idaes.__file__).parent
-    failures, total = importr(root_dir)
-    n = len(failures)
-    assert n == 0, f"{n:d} failures in {total:d} tests: {failures}"
+@pytest.mark.parametrize("module_path", list(idaes_path.rglob("*.py")))
+def test_toplevel_statements(module_path):
+    """Look for "too many" top-level statements.
+
+    This test is far quicker than `test_import`, but they test different things.
+    It does not have any way of knowing how long it takes to import the module
+    (or whether it can be imported).
+
+    It also has a pretty crude idea of what a top-level statement is (and isn't)
+    """
+    # Skip some paths
+    if module_path == idaes_path / "__init__.py":
+        return
+
+    with module_path.open(encoding="utf8") as f:
+        num_statements = count_statements(f)
+        assert num_statements <= TOPLEVEL_LINES_MAX
+
+
+def count_statements(f):
+    statements = 0
+    comment_block = False
+    lineno = 0
+
+    # Look at file line-by-line
+    for line in f:
+        lineno += 1
+        is_statement = False
+        # if in a comment block, wait for it to end
+        if comment_block:
+            if re.match(r".*('''|\"\"\")", line):
+                comment_block = False
+        # look for start of comment block
+        elif re.match(r"\s*('''|\"\"\")", line):
+            comment_block = True
```

Comment:
This will get confused by:
```python
def foo():
    """"a docstring"""
```

Also, wrt vernacular, `"""` is not a comment block, but a multiline string.

Reply:
I mean, yes, but who would do that? I guess it could be more robust though.

Reply:
Oh, about 144 files in IDAES, including
- `./idaes/__init__.py`
- `./idaes/dmf/magics.py`
- ...

Reply:
I think the following would be more robust:
```python
comment_block ^= bool(len(re.findall(r".*('''|\"\"\")", line)) % 2)
if comment_block:
    continue
```
...it will still get confused by blockquotes that contain blockquotes (i.e., putting `'''` inside a blockquote delimited by `"""`)... but the logic to catch that starts to get rather challenging.